TM1py/Services/CubeService.py

```diff
@@ -249,6 +250,39 @@ def search_for_dimension_substring(self, substring: str, skip_control_cubes: boo
         cube_dict = {entry['Name']: [dim['Name'] for dim in entry['Dimensions']] for entry in response.json()['value']}
         return cube_dict
 
+    def enable_cube_rule(self, cube: Cube) -> None:
+        """
+        Enable a cube rule from its base64-encoded hash if it exists.
+
+        :param cube: An instance of a Cube.
+        """
+        current_rule = cube.rules.text
+        if not current_rule:
+            # If there is no rule, there is nothing to do.
+            return
+        try:
+            current_rule = current_rule[1:] if current_rule.startswith('#') else current_rule
+            cube.rules = base64.b64decode(current_rule).decode('utf-8')
+        except Exception:
+            raise ValueError(f"Current rule is not decodable by b64decode standards")
```

Comment:
I think RuntimeError would be applicable here.

In TM1py the ValueError is mostly used for invalid arguments

Reply:
fixed in comit #f6ee7209d4b830bc4a0e8b733508bc27bc9f4490